src/Centreon/Application/Controller/Monitoring/CommentController.php

```diff
@@ -0,0 +1,346 @@
+            ) {
+                continue;
+            }
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Entry point to add comments to multiple resources
+     *
+     * @param Request $request
+     * @return View
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     */
+    public function addResourcesComment(
+        Request $request
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+        /**
+        * @var Contact $contact
+        */
+        $contact = $this->getUser();
+        $this->commentService->filterByContact($contact);
+
+       /*
+        * Validate the content of the POST request against the JSON schema validator
+        */
+        $results = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/CommentResources.json'
+        );
+
+        /**
+         * If user has no rights to add a comment for host and/or service
+         * return view with unauthorized HTTP header response
+         */
+        if (!$this->hasCommentRightsForResources($contact, $results['resources'])) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        /**
+         * Dissect the results to extract the hostids and serviceids from it
+         */
+        $hostIds = [];
+        $serviceIds = [];
+        $comments = [];
+
+        $now = new \DateTime('now');
+
+        foreach ($results['resources'] as $index => $commentResource) {
+            $date = ($commentResource['date'] !== null) ? new \DateTime($commentResource['date']) : $now;
+            $comments[$index] = (new Comment($commentResource['id'], $commentResource['comment']))
+                ->setDate($date)
+                ->setParentResourceId($commentResource['parent']['id']);
+
+            if ($commentResource['type'] === ResourceEntity::TYPE_HOST) {
+                $hostIds[$index] = $commentResource['id'];
+            } elseif ($commentResource['type'] === ResourceEntity::TYPE_SERVICE) {
+                $serviceIds[$index] = [
+                    'host_id' => $commentResource['parent']['id'],
+                    'service_id' => $commentResource['id']
+                ];
+            }
+        }
+
+        /**
+         * Retrieving all services and hosts
+         */
+        $hosts = $this->monitoringService->findMultipleHosts($hostIds);
+        $services = $this->monitoringService->findMultipleServices($serviceIds);
+
+        if (!empty($hosts)) {
+            foreach ($hosts as $key => $host) {
+                $this->commentService->addHostComment($comments[$key], $host);
+            }
+        }
+
+        if (!empty($services)) {
+            foreach ($services as $key => $service) {
+                $this->commentService->addServiceComment($comments[$key], $service);
+            }
+        }
```

Comment:
if (!empty($array))  is not needed, as if your $array is empty you will not go into your foreach loop.

Reply:
Yep thanks you're right :+1: I'll modify this

Reply:
done :heavy_check_mark: